src/libcore/iter/traits/iterator.rs

```diff
@@ -2742,7 +2742,7 @@ pub trait Iterator {
             None => return true,
         };
 
-        while let Some(curr) = self.next() {
+        for curr in self {
```

Comment:
For comparison to the previous comment, this one seems fine, since it can move `self` and doesn't add extra `&mut`s.  (Though it should arguably be rewritten as a `try_fold`...)

Reply:
Thank you for explanation.
I cannot figure out how to change it to `try_fold` so should I keep it or revert?

Reply:
I would revert.

Reply:
IIRC generated assembly was exactly the same but I'll drop it anyway.
Should I add FIXME about rewriting it with `try_fold`?